### PR TITLE
Add getThreadPictures implementation

### PR DESCRIPTION
**Pull request by enhdless**
Basic implementation for [#288](https://github.com/Schmavery/facebook-chat-api/issues/288)

### CODE REVIEW

**Comment by GAMELASTER** [comment]
whoah, nice job! I will try it!

**Comment by Schmavery** [comment]
@GAMELASTER any thoughts after trying it?
@bsansouci how does the code look to you?  Too bad about the messy response but I guess it works for now...?

File: src/getThreadPictures.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    };
+
+    defaultFuncs
+      .post("https://www.facebook.com/ajax/messaging/attachments/sharedphotos.php", ctx.jar, form)
+      .then(utils.parseAndCheckLogin(ctx.jar, defaultFuncs))
+      .then(function(resData) {
+        if (resData.error) {
+          throw resData;
+        }
+        return Promise.all(resData['payload']['imagesData'].map(function(image) {
```
Comment (bsansouci) [inline src/getThreadPictures.js]: use `resData.payload.imagesData`

File: src/getThreadPictures.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+            defaultFuncs
+            .post("https://www.facebook.com/ajax/messaging/attachments/sharedphotos.php", ctx.jar, form)
+            .then(utils.parseAndCheckLogin(ctx.jar, defaultFuncs))
+            .then(function(resData) {
+              if (resData.error) {
+                throw resData;
+              }
+              // the response is pretty messy
+              var queryThreadID = resData.jsmods.require[0][3][1].query_metadata.query_path[0].message_thread;
+              var imageData = resData.jsmods.require[0][3][1].query_results[queryThreadID].message_images.edges[0].node.image2;
```
Comment (bsansouci) [inline src/getThreadPictures.js]: Can you explain what `edges` and `image2` are?

File: src/getThreadPictures.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+            form = {
+              'thread_id': threadID,
+              'image_id': image.fbid
+            };
+            defaultFuncs
+            .post("https://www.facebook.com/ajax/messaging/attachments/sharedphotos.php", ctx.jar, form)
+            .then(utils.parseAndCheckLogin(ctx.jar, defaultFuncs))
+            .then(function(resData) {
+              if (resData.error) {
+                throw resData;
```
Comment (bsansouci) [inline src/getThreadPictures.js]: I'm don't know promises super well, but don't know need to call `reject` here?

File: src/getThreadPictures.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
+"use strict";
+
+var utils = require("../utils");
+var log = require("npmlog");
+
+module.exports = function(defaultFuncs, api, ctx) {
+  return function getThreadPictures(threadID, offset, limit, callback) {
+    if(!callback) {
```
Comment (bsansouci) [inline src/getThreadPictures.js]: Don't make callback optional, it wouldn't make any sense not to give a callback.

File: src/getThreadPictures.js (hunk 2 above)
Comment (enhdless) [inline src/getThreadPictures.js]: I'm not sure about `edges`, it is just part of the response:
```
"edges": [{
    "node": {
        "id": "10207439106202914",
        "image1": {
            "uri": "https:\/\/scontent.xx.fbcdn.net\/v\/t34.0-12\/13734559_10207439106202914_2090663689_n.png?oh=879662957f0cc6733705ccc45fa8c242&oe=579A5ECD",
            "width": 960,
            "height": 355
        },
        "image2": {
            "uri": "https:\/\/scontent.xx.fbcdn.net\/v\/t35.0-12\/s2048x2048\/13833333_10207439106202914_2090663689_o.png?oh=6b4e847e5de29530f5a7e87d2e52f082&oe=579956A8",
            "width": 2048,
            "height": 757
        }
    },
    "cursor": "10207439106202914"
}],
```
As for `image2`, there always seems to be `image1` and `image2` returned in `node`, but `image2` is the one with the greater dimensions and correct original size, which is why I chose to return that value.

File: src/getThreadPictures.js (hunk 3 above)
Comment (enhdless) [inline src/getThreadPictures.js]: Turns out I was complicating it by wrapping it in a new promise (`return new Promise`). Fixed code in commit.

**Comment by bsansouci** [comment]
Hey @enhdless does this still work :p 
If yes then I think we could merge.

**Comment by ravkr** [comment]
any updates?

**Comment by Schmavery** [comment]
@ravkr Any chance you could sync to this branch and test if the feature works on your machine?  If it does, I think we can merge :)
It's just nice to have a couple of data points verifying that a feature works for everyone before adding it.

**Comment by enhdless** [comment]
Oops, let this slide under my radar. Will take a look later today!

**Comment by Schmavery** [comment]
Awesome, sounds great @enhdless

**Comment by enhdless** [comment]
Still works great for me. I'll make the documenation a bit more clear, squash the commits, and then it should be ready to go.

**Comment by enhdless** [comment]
If anyone else could test this then that'd be great, but otherwise I think this can be merged.

**Comment by ravkr** [comment]
Works well for me :D
but is it possible to get sender, timestamp etc.?
like on messenger.com :/
messenger1
messenger2

**Comment by Schmavery** [comment]
I'm going to merge this but to submit a PR adding the functionality you want @ravkr!  Maybe @enhdless has some tips for you.

**Pull request merged**